### DISCUSSION TITLE
How to UpdateStatus and requeue "Reconcile" events until a desired state is reached?

### DISCUSSION THREAD

**Discussion by Karthik2893** [dotnet/dotnet-operator-sdk#550] [Ideas & General Discussion]
I have a CRD which defines a resource (say X). CRUD operations on X are handled by a different service which returns a "requestId" when a resource creation is requested. Following are the sequence of operations I am currently doing in my controller:
1. When a custom-resource is created, I am calling the service to create the resource.
2.  Updating the "Status" to reflect the "requestId" and the "state" of X. 
3. If the "state" != "Succeeded", requeue "Reconcile" event with a delay.
4. If state == "succeeded" , return null.

In my observation, whenever I add "Step 2",  the "Reconcile" action/method is not being invoked on the controller. 

Am I doing something of an anti-pattern here? 

Some due diligence (incase it might help anyone looking in anyway):
I have a very basic understanding of reactive streams, so please bear with me.  My understanding from this code is that we are grouping events for each individual resource and by  invoking "Switch", only projecting the recent observables? Could this be causing this by any chance?
https://github.com/buehler/dotnet-operator-sdk/blob/3f42bbcb13d98a0811606b4b8ca83566c74a988a/src/KubeOps/Operator/Controller/EventQueue.cs#L44-L55

Thanks a lot!

**Comment by buehler** [comment]
Discussed in https://github.com/buehler/dotnet-operator-sdk/discussions/551 :-)
